tests/host_tools/metrics.py

```diff
@@ -501,24 +502,31 @@ def validate(self, microvm):
 
         # make sure "{self.name}" is aggregate of "{self.name}_*"
         # and that there are only {num_dev} entries of "{self.name}_*"
-        self.validate_aggregation(fc_metrics)
+        self.validate_per_device_metrics(fc_metrics)
         print(f"\nsuccessfully validated aggregate of {self.dev_name} metrics")
 
-    def validate_aggregation(self, fc_metrics):
+    def validate_per_device_metrics(self, fc_metrics):
         """
         validate aggregation of device metrics
         """
-        metrics_aggregate = fc_metrics[self.dev_name]
         metrics_calculated = {}
         actual_num_devices = 0
         print(f"In aggregation of {self.dev_name} expected {self.num_dev=}")
         for component_metric_names, component_metric_values in fc_metrics.items():
-            if f"{self.dev_name}_" in component_metric_names:
-                print(f"found {component_metric_names} during aggr of {self.dev_name}")
+            if (
+                f"{self.dev_name}_" in component_metric_names
+                and component_metric_names.startswith(self.dev_name)
+            ):
+                print(
```

Comment:
I appreciate this has already been there, but is there a reason we `print()` something in a test?

Reply:
this was useful in debugging locally so kept it to provide more data if a breaking change is detected. But I can remove this as a breaking change would still have to be debugged and tested locally anyways.

Reply:
I have no strong opinion here. I think we should aim for printing all the useful information in the asserts, but if it is not possible here, I'm ok to leave it as is.

Reply:
prints were useful for my local debug but they don't need to be in the PR so removed them.